api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/TenantsHandler.java

```diff
@@ -139,4 +142,22 @@ public void deleteTenant(@Suspended AsyncResponse asyncResponse, @PathParam("id"
                 }
         );
     }
+
+    @GET
+    @Path("/metrics")
+    @ApiOperation(value = "Returns a list of all metric definitions in the system.", response = Metric.class,
+            responseContainer = "List")
+    public void findAllDefinitions(@Suspended AsyncResponse asyncResponse) {
```

Comment:
Do we have to worry about this method as well as scanAllMetricIndexes causing heap pressure for larger data sets?

Reply:
It might, yes. Although it's executed only once when Heapster restarts, so there's no other load in the system at that time.